File: articles/computational-notebooks-reproducible-biological-research/julia/reproducibility_kernel.jl
# Compact reproducibility kernel in Julia.
# This dependency-free example calculates workflow completeness.

required_steps = 6
documented_steps = 6
failed_cells = 0
executed_cells = 4

workflow_completeness = documented_steps / required_steps
failure_rate = failed_cells / executed_cells

println("workflow_completeness=", round(workflow_completeness, digits=5))
println("failure_rate=", round(failure_rate, digits=5))

if workflow_completeness == 1.0 && failure_rate == 0.0
    println("status=pass")
else
    println("status=review")
end

File: articles/computational-notebooks-reproducible-biological-research/fortran/reproducibility_kernel.f90
program reproducibility_kernel
  implicit none

  integer :: required_steps
  integer :: documented_steps
  integer :: failed_cells
  integer :: executed_cells
  real :: completeness
  real :: failure_rate

  required_steps = 6
  documented_steps = 6
  failed_cells = 0
  executed_cells = 4

  completeness = real(documented_steps) / real(required_steps)
  failure_rate = real(failed_cells) / real(executed_cells)

  print *, "workflow_completeness=", completeness
  print *, "failure_rate=", failure_rate

  if (completeness == 1.0 .and. failure_rate == 0.0) then
     print *, "status=pass"
  else
     print *, "status=review"
  end if

end program reproducibility_kernel
